### PR TITLE
Add StreamIdManager

### PR DESCRIPTION
**Pull request by bryce-anderson**
Introduce the accounting abstraction for stream ids.

### CODE REVIEW

**Comment by bryce-anderson** [comment]
I'll plan on merging this sometime this weekend unless there are objections.

File: http/src/main/scala/org/http4s/blaze/http/http2/StreamIdManager.scala [hunk 1]
```diff
@@ -0,0 +1,93 @@
+package org.http4s.blaze.http.http2
+
+private object StreamIdManager {
+
+  /** Create a new [[StreamIdManager]] */
+  def apply(isClient: Boolean): StreamIdManager
+    = create(isClient, 0)
```
Comment (rossabaker) [inline http/src/main/scala/org/http4s/blaze/http/http2/StreamIdManager.scala]: This is an odd indentation of the `=`.

**Comment by bryce-anderson** [comment]
@rossabaker, no worries, I appreciate the review.

File: http/src/main/scala/org/http4s/blaze/http/http2/StreamIdManager.scala (hunk 1 above)
Comment (bryce-anderson) [reply http/src/main/scala/org/http4s/blaze/http/http2/StreamIdManager.scala]: Indeed, I'll fix it.

**Pull request merged**